dotCMS/src/main/java/com/dotcms/rest/api/v1/apps/AppsHelper.java

```diff
@@ -164,24 +176,65 @@ Response getAppSiteView(
             if (null == host) {
-                throw new DotDataException(
-                        String.format(" Couldn't find any host with identifier `%s` ", siteId));
+                throw new DoesNotExistException(
+                      String.format(" Couldn't find any host with identifier `%s` ", siteId)
+                );
             }
 
             final Optional<AppSecrets> optionalAppSecrets = appsAPI
                     .getSecrets(key, true, host, user);
-            if (optionalAppSecrets.isPresent()) {
-                final AppSecrets appSecrets = protectHiddenSecrets(
-                        optionalAppSecrets.get());
-                final SiteView siteView = new SiteView(host.getIdentifier(), host.getHostname(),
-                        appSecrets.getSecrets());
-                return Optional.of(new AppView(appDescriptor, 1L,
-                        ImmutableList.of(siteView)));
-            }
+
+                //We need to return a view with all the secrets and also descriptors of the remaining parameters merged.
+                //So we're gonna need a copy of the params on the yml.
+                final Map<String, ParamDescriptor> descriptorParams = new HashMap<>(appDescriptor.getParams());
+                //First will process the secrets stored..
+                //As we process them we we remove them from the `descriptorParams` map.
+                //They're removed from the map as we go on so we know that what's left in the map doesnt have a secret in storage.
+                final AppSecrets appSecrets = optionalAppSecrets.isPresent() ? protectHiddenSecrets(optionalAppSecrets.get()) : AppSecrets.empty() ;
+                final Set<SecretView> mappedSecrets = appSecrets.getSecrets().entrySet()
+                        .stream()
+                        .map(e -> new SecretView(e.getKey(),e.getValue(), descriptorParams.remove(e.getKey()))).collect(
+                                Collectors.toSet());
+
+                //Now we process the remaining on `descriptorParams`.
+                //Transform the DescriptorParams into SecretView
+                //What ever is left in there is a param that exist on the yml.
+                // that doesnt have a secret in storage.
+                final Set<SecretView> mappedDescriptors = appDescriptor.getParams().keySet().stream()
+                        .map(
+                                paramKey -> new SecretView(paramKey, null,
+                                        descriptorParams.remove(paramKey))).collect(
+                                Collectors.toSet());
+
+                //At this point `descriptorParams` should be empty.
+                assert (descriptorParams.isEmpty());
+
+                //Now we need to present them both.
+                //For which we first add the ones from the descriptor.
+                final Set<SecretView> merged = new LinkedList<>(mappedDescriptors).stream()
```

Comment:
not sure why do we need the new List if you are going to sort at the end again

Reply:
yeah, there was a reason for that..  I think I was trying to get away from a concurrent modification exception But let me check if that can go away

Reply:
you were right no need to new up the list before filtering and sorting. Done